### PR TITLE
Separate operator and envd helm charts

### PR DESCRIPTION
**Pull request by bobbyiliev**
As discussed, this PR separates the Materialize Operator and Environmentd into two separate Helm charts.

Also adding the latest changes as per https://github.com/MaterializeInc/materialize/pull/30090

### CODE REVIEW

File: misc/helm-charts/environmentd/values.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    environmentdImageRef: "materialize/environmentd:v0.119.2"
+    environmentdExtraArgs:
+      - "--log-filter=info"
+    environmentdCpuAllocation: "1"
+    environmentdMemoryAllocation: "1Gi"
+    requestRollout: 22222222-2222-2222-2222-222222222222
+    forceRollout: 33333333-3333-3333-3333-333333333333
+    inPlaceRollout: false
+    secret:
+      name: "materialize-backend-12345678-1234-1234-1234-123456789012"
```
Comment (doy-materialize) [inline misc/helm-charts/environmentd/values.yaml:27]: the secret name should just be `materialize-backend-${environment_name}`, it's not configurable within orchestratord

File: misc/helm-charts/environmentd/values.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      metadataBackendUrl: "postgres://user:pass@cockroachdb-public.cockroachdb.svc.cluster.local:26257/db?sslmode=verify-full&sslrootcert_inline=ca_cert"
+      persistBackendUrl: "s3://minio:minio123@bucket/12345678-1234-1234-1234-123456789012?endpoint=http%3A%2F%2Fminio.minio.svc.cluster.local%3A9000&region=minio"
+
+# Additional environments can be added here
+
+# Global settings
+global:
+  operatorNamespace: "materialize-system"
```
Comment (doy-materialize) [inline misc/helm-charts/environmentd/values.yaml:35]: what is this used for?

File: misc/helm-charts/operator/templates/_helpers.tpl [hunk 3]
```diff
@@ -100,3 +100,10 @@ Create the Metadata database connection string
 {{- define "materialize-operator.metadatadbConnectionString" -}}
 postgresql://{{ .Values.metadatadb.username }}:{{ .Values.metadatadb.password }}@{{ .Values.metadatadb.endpoint }}:{{ .Values.metadatadb.port }}/defaultdb?sslmode=verify-full&sslrootcert=/metadata/metadata-certs/ca.crt
 {{- end }}
+
+{{/*
+Create the name of the namespace to use
+*/}}
+{{- define "materialize-operator.namespaceName" -}}
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply misc/helm-charts/operator/templates/_helpers.tpl:107]: it's a helper function, it is used in the templates and allows users to set a custom namespace for the operator installation, with a fallback to the release namespace if none is specified.

File: misc/helm-charts/environmentd/values.yaml (hunk 2 above)
Comment (bobbyiliev) [reply misc/helm-charts/environmentd/values.yaml]: ah that was indeed not used anywhere

File: misc/helm-charts/environmentd/values.yaml (hunk 1 above)
Comment (bobbyiliev) [reply misc/helm-charts/environmentd/values.yaml]: ok nice! thanks!

File: misc/helm-charts/environmentd/values.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Default namespace for Materialize Environmentd
+namespace:
+  create: true
+  name: "materialize-environment"
+
+# Default values for Materialize Environmentd chart
+materializeEnvironments:
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply misc/helm-charts/environmentd/values.yaml]: Now that you mention it, it actually probably better practice to manage a single environment per chart installation. Just changed that

**Review by doy-materialize — APPROVED** [review]

**Pull request merged**